File: Stack/DailyTemperatures.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> dailyTemperatures(vector<int> &temperatures)
{
    vector<int> ans(temperatures.size(), 0);
    stack<int> stk;

    for (int i = 0; i < temperatures.size(); i++)
    {
        while (!stk.empty() && temperatures[i] > temperatures[stk.top()])
        {
            ans[stk.top()] = i - stk.top();
            stk.pop();
        }

        stk.push(i);
    }

    return ans;
}

File: Stack/DailyTemperatures.py
from typing import List


class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0] * len(temperatures)
        stack = []

        for i in range(len(temperatures)):
            while len(stack) > 0 and temperatures[i] > temperatures[stack[-1]]:
                j = stack.pop()
                res[j] = i - j

            stack.append(i)

        return res
